File: ysda/1_1/d_squares_n_cubes.py
def solve(n):
    s = set()
    i = 1
    while i * i * i <= n:
        s.add(i * i * i)
        i += 1
    i = 1
    while i * i <= n:
        s.add(i * i)
        i += 1
    return len(s)

t = int(input())
ans = []
for i in range(t):
    n = int(input())
    ans.append(solve(n))

for i in ans:
    print(i)

File: ysda/1_1_sets_n_dicts/d_squares_n_cubes.cpp
#include <bits/stdc++.h>

using namespace std;

long long solve(long long n) {
	set<long long> s;
	for (long long i = 1; i * i * i <= n; ++i) {
		s.insert(i * i * i);
	}
	for (long long i = 1; i * i <= n; ++i) {
		s.insert(i * i);
	}
	return s.size();
}

int main() {
	int t;
	long long n;
	cin >> t;
	long long ans[t];
	for (int i = 0; i < t; ++i) {
		cin >> n;
		ans[i] = solve(n);
	}
	for (int i = 0; i < t; ++i) {
		cout << ans[i] << '\n';
	}

	return 0;
}
